### PR TITLE
Add support for SRR of automatic groups

### PR DESCRIPTION
**Pull request by stark-dev**
Signed-off-by: Mauro Guerrera <mauroguerrera@eaton.com>

### CODE REVIEW

**Comment by stark-dev** [comment]
@etnKlendathu , @perrettecl please double check the control file ;)

File: common/common/srr.h [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+namespace fty::common::srr {
+
+static constexpr const char* Channel       = "FTY.Q.GROUP.SRR";
+static constexpr const char* FeatureName   = "automatic-groups";
+static constexpr const char* ActiveVersion = "1.0";
+
+struct SrrGroupObj : public pack::Node
+{
+    pack::ObjectList<Group> m_groupList = FIELD("groupList");
```
Comment (etnKlendathu) [inline common/common/srr.h:15]: when you use pack struct you always use a field as getter/setter member.
srrgroup.groupList....
so, m_ prefix is somehow for private/protected members

File: server/src/lib/server.cpp [hunk 2]
```diff
@@ -45,7 +60,7 @@ void Server::reloadConfig()
 
 void Server::process(const Message& msg)
 {
-    logDebug("Automatic group: got message {}, payload:\n   {}", msg.meta.subject.value(), msg.userData.asString());
+    logDebug("Automatic group: got message {}", msg.dump());
```
Comment (etnKlendathu) [inline server/src/lib/server.cpp:63]: :(

**Review by etnKlendathu — CHANGES_REQUESTED** [review]
This is quite ok, but one big comment...
You did a job as not a job as others (other jobs are thread job, each routine pushed to the pool and server is free to accept other stuff). You did that all srr functions will block main server thread. I not sure if we want it.

File: server/src/lib/task.h [hunk 3]
```diff
@@ -124,24 +124,21 @@ class Task : public fty::Task<T>
...
+    Task() = default;
 
     void operator()() override
     {
         Response<ResponseT> response;
         try {
             if (auto it = dynamic_cast<T*>(this)) {
                 if constexpr (!std::is_same<InputT, void>::value) {
-                    if (m_in.userData.empty()) {
+                    if (m_in.userData[0].empty()) {
```
Comment (etnKlendathu) [inline server/src/lib/task.h:135]: very risky... you should check that data have at least one element (!empty)

File: server/src/lib/task.h [hunk 4]
```diff
@@ -66,18 +66,18 @@ class Response : public pack::Node
 
         if (status == Message::Status::Ok) {
             if (out.hasValue()) {
-                msg.userData.setString(*pack::json::serialize(out));
+                msg.userData.append(*pack::json::serialize(out));
```
Comment (etnKlendathu) [inline server/src/lib/task.h:69]: hmmm... to be honest, i don't like append here. What if some content already exists in data? Better to make replace here.

File: server/src/lib/server.h [hunk 5]
```diff
@@ -1,8 +1,13 @@
 #pragma once
 #include "common/message-bus.h"
+#include <fty_common_dto.h>
 #include <fty/event.h>
 #include <fty/thread-pool.h>
 
+namespace messagebus {
```
Comment (etnKlendathu) [inline server/src/lib/server.h:7]: I think you don't need it here already.

**Review by etnKlendathu — CHANGES_REQUESTED** [review]
quite ok :)
I mostly don't like to use append... you could add method to message setUserData() with clear call

**Pull request merged**